### DISCUSSION TITLE
Access relation's data within $this->data within a form

### DISCUSSION THREAD

**Discussion by invaders-xx** [filamentphp/filament#3552] [Help] [answered]
I am facing an issue with version 2.15.19 with repeater using a relation. Before (up to 2.15.18) I was able to access the relation's information (using a repeater) via $this->data in the form. Since 2.15.19 update,  I have an error in accessing $this->data['relation'] in edit and view page within a resource. Maybe it comes form PR #3495 

@danharrin , any thoughts ? any changes in that PR which may causes this breaking change ?

**Comment by danharrin** [comment, marked as answer]
Ah right, yeah. So it was never intended that relations were loaded as part of `$this->data`, since this is *form data only that gets saved to the model*, not loaded eloquent relationships that just happened to be loaded when the form is filled. We switched from using `$model->toArray()` to `$model->attributesToArray()`, which provides a performance improvement since we don't serialise relationships any more.

The recommended way to access relationships is (and has always been) `$this->record->relationNameHere`

**Reply by invaders-xx** [reply]
@danharrin, I agree with you, my only concern is with Create as $this->record is null

**Reply by danharrin** [reply]
Yeah, but you don't have relations on non-existant records anyway?

**Reply by danharrin** [reply]
And you specifically mentioned only edit & view in your original post

**Reply by invaders-xx** [reply]
My apologies, I forgot to mention create, I did in discord, my bad. Previously I was able to manipulate $this->data with relations even for create, edit and view. I will try to think/implement in another way and will come back to you if I am stuck.

**Reply by danharrin** [reply]
I am confused, relations were not available on create, as the record doesnt exist?

**Reply by invaders-xx** [reply]
I was talking about hasMany relations, in a repeater for example

**Reply by danharrin** [reply]
`Repeater::make()->relationship()` should still work fine?
